northd/northd.c

```diff
@@ -8679,12 +8692,63 @@ build_lswitch_arp_nd_responder_known_ips(struct ovn_port *op,
                     "outport = inport; "
                     "flags.loopback = 1; "
                     "output;",
-                    op->peer->lrp_networks.ea_s,
-                    op->peer->lrp_networks.ea_s);
+                    ea_s,
+                    ea_s);
 
                 ovn_lflow_add_with_hint(lflows, op->od, S_SWITCH_IN_ARP_ND_RSP,
                     50, ds_cstr(match), ds_cstr(actions), &op->nbsp->header_);
             }
+
+            /* Add IPv6 NDP responses.
+             * For ND solicitations, we need to listen for both the
+             * unicast IPv6 address and its all-nodes multicast address,
+             * but always respond with the unicast IPv6 address. */
+            if (proxy_arp_addrs.n_ipv6_addrs) {
+                struct ds ip6_dst_match = DS_EMPTY_INITIALIZER;
```

Comment:
Using `struct ds` here is completely fine, however don't forget to free them afterwards with `ds_destroy`. Probably before the if ends is best fit.